shell/src/main/java/alluxio/cli/bundler/command/CollectLogCommand.java

```diff
@@ -14,21 +14,148 @@
+  public static final Set<String> FILE_NAMES_PREFIXES = Stream.of(
+      "master.log",
+      "master.out",
+      "job_master.log",
+      "job_master.out",
+      "master_audit.log",
+      "worker.log",
+      "worker.out",
+      "job_worker.log",
+      "job_worker.out",
+      "proxy.log",
+      "proxy.out",
+      "task.log",
+      "task.out",
+      "user"
+  ).collect(Collectors.toSet());
+  // We tolerate the beginning of a log file to contain some rows that are not timestamped.
+  // A YARN application log can have >20 rows in the beginning for
+  // general information about a job.
+  // The timestamped log entries start after this general information block.
+  private static final int TRY_PARSE_LOG_ROWS = 100;
+
+  // Preserves the order of iteration, we try the longer pattern before the shorter one.
+  // The 1st field is the DateTimeFormatter of a specific pattern.
+  // The 2nd field is the length to take from the beginning of the log entry.
+  // The length of the format string can be different from the datetime string it parses to.
+  // For example, "yyyy-MM-dd'T'HH:mm:ss.SSSXX" has length of 27 but parses to
+  // "2020-10-12T12:11:10.055+0800".
+  // Note that the single quotes around 'T' are not in the real string,
+  // and "XX" parses to the timezone, which is "+0800".
+  // The datetime parsing works only when the string matches exactly to the format.
+  private static final Map<DateTimeFormatter, Integer> FORMATTERS =
+          new LinkedHashMap<DateTimeFormatter, Integer>(){
+    {
+      // "2020-01-03 12:10:11,874"
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss,SSS"), 23);
+      // "2020-01-03 12:10:11"
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"), 19);
+      // "2020-01-03 12:10"
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"), 16);
+      // "20/01/03 12:10:11"
+      put(DateTimeFormatter.ofPattern("yy/MM/dd HH:mm:ss"), 17);
+      // "20/01/03 12:10"
+      put(DateTimeFormatter.ofPattern("yy/MM/dd HH:mm"), 14);
+      // 2020-01-03T12:10:11.874+0800
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXX"), 28);
+      // 2020-01-03T12:10:11
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss"), 19);
+      // 2020-01-03T12:10
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm"), 16);
+    }
+  };
+
+  private String mLogDirPath;
+  private File mLogDir;
+  private URI mLogDirUri;
+  private Set<String> mIncludedPrefix;
+  private Set<String> mExcludedPrefix;
+  private LocalDateTime mStartTime;
+  private LocalDateTime mEndTime;
+
+  public static final String INCLUDE_OPTION_NAME = "include-logs";
+  public static final String EXCLUDE_OPTION_NAME = "exclude-logs";
+  public static final String ADDITIONAL_OPTION_NAME = "additional-logs";
+  private static final Option INCLUDE_OPTION =
+          Option.builder().required(false).argName("filename-prefixes")
```

Comment:
I think maybe we can achieve everything with only the `--include` flag, but that flag simply specifies the files to get. We can set the default for this flag as the list of `FILE_NAMES_PREFIXES`.

However, if a user wants to exclude or include files, then they would have to add a `--include` with the full list they want.

If that is too unfriendly, then can't we just assume we download everything under `logs/`, and only have an `--exclude` flag, so any additional logs will automatically get retrieved, and the exclude is a simple way to not download some.

I just thought having 3 flags was too much. Ideally we only need one. What do you think?

Reply:
I guess keeping only the `--include` will make it too verbose. For example I want everything plus my gc logs, then I will have to type `--include-logs master,worker,job-master,job-worker,user,task,alluxio_gc`. For less experienced users we may need to teach them how to make sure all components are included, to avoid another iteration. This makes `--additional` useful. For similar argument I would suggest keeping `--exclude` too.

On your other option, where we keep everything unless explicitly excluded. I've seen cases where some users put many irrelevant logs under Alluxio log dir and `collectInfo` is run by a general L1 team that just follow handbooks without looking into the Alluxio log dir. Then hundreds of blue talon and ZK logs are copied (10M * 100 for each node) and the collectInfo tarball size went to the ceiling. I hope to avoid this kind of pitfall.

Reply:
Ahhh, I see. I didn't realize there would be so many extra logs like zk or others.

Why do we need 3 flags? Can't we just have the internal list of what we know, and just 2 flags `--exclude` and `--additional`?

Reply:
Logically we can remove the `--include` flag but that will make it a bit harder to just keep logs for one or two components. We have six different groups of log/out files `master,worker,job-master,job-worker,user,task` and I don't expect users to be very clear about what to exclude.

It gets trickier if we want finer granularity, like "get the audit logs", which translates to `--include-logs master_audit` or `--exclude master.log, master.out, job-master, worker, job-worker, task, user`.

Reply:
Ok, for now, this should be fine.